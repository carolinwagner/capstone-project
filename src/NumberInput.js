import React, { forwardRef } from 'react'
import styled from 'styled-components/macro'

const NumberInput = forwardRef(({ question }, ref) => {
  return (
    <StyledNumberInput
      type="number"
      ref={ref}
      name={question?.name || 'defaultTextInput'}
    />
  )
})

export default NumberInput

const StyledNumberInput = styled.input`
  width: 25%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid var(--lightblue);
  border-radius: 4px;
`
