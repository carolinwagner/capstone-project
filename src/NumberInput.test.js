import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import NumberInput from './NumberInput'

describe('NumberInput', () => {
  it('renders NumberInput correctly', () => {
    const tree = renderer.create(<NumberInput />)
    expect(tree).toMatchSnapshot()
  })
})
