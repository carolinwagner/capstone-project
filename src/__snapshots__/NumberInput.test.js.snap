// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput renders NumberInput correctly 1`] = `undefined`;

exports[`NumberInput renders correctly 1`] = `undefined`;
